docs/src/advanced-concepts/auxiliary-outputs.rst

```diff
@@ -0,0 +1,24 @@
+Auxiliary outputs
+=================
+
+These outputs, which are idenfified by the ``mts-models::aux::`` prefix,
+represent additional information that the model may provide. They are not
+conventional trainable outputs, and they often correspond to internal
+information that the model is capable of providing, such as its internal
+representation.
+
+The following auxiliary outputs that are currently supported
+by one or more architectures in the library:
+
+- ``mts-models::aux::last_layer_features``: The internal representation
+   of the model at the last layer, before the final linear transformation.
+
+The following table shows the architectures that support each of the
+auxiliary outputs:
+
++-----------------------------------------+-----------+------------------+-----+
+| Auxiliary output                        | SOAP-BPNN | Alchemical model | PET |
++=========================================+===========+==================+=====+
+| `mts-models::aux::last_layer_features`  | Yes       |       No         | No  |
```

Comment:
For consistency, I would use all dash or all underscores.

```suggestion
| `mts-models::last-layer-features`  | Yes       |       No         | No  |
```

```suggestion
| `mts_models::last_layer_features`  | Yes       |       No         | No  |
```

(I'm alo not sure what the `aux` gives us here)

Reply:
This should also be a double \`, not a single one

Reply:
Thanks for the formatting. Regarding `aux`, it essentially discriminates between trainable and non-trainable outputs of the model. I think it's very useful, but we can discuss if you're around

Reply:
ok, I see you point for `aux`. I'm not sure how to best convey this to contributors. Do you foresee other similar outputs?

Reply:
Perhaps the name is not the best (although it would be easy to change later as long as we establish that the concept is useful). Yes, I can see some models returning arbitrary data that should not be considered as proper predictions. I'm thinking of other internal representations (not necessarily last-layer), symmetrization errors (for PET in the case where you average across a few orientations), uncertainties, perhaps timings, perhaps fictitious charges in mesh-LODE (still kind of an internal representation), etc